### PR TITLE
Account for _tagged_ dollar-quoted strings when stripping unicode spaces

### PR DESCRIPTION
**Pull request by lnkuiper**
Follow-up of #12405

### CODE REVIEW

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - some comments:

File: src/parser/parser.cpp [hunk 1]
```diff
@@ -96,14 +106,22 @@ bool Parser::StripUnicodeSpaces(const string &query_str, string &new_query) {
 			quote = query[pos];
 			pos++;
 			goto in_quotes;
-		} else if (query[pos] == '$' && (query[pos + 1] < '0' || query[pos + 1] > '9')) {
-			// (optionally tagged) dollar-quoted string, which can't start with a digit
+		} else if (query[pos] == '$' &&
+		           (query[pos] == '$' || IsValidDollarQuotedStringTagFirstChar(static_cast<char>(query[pos])))) {
```
Comment (Mytherin) [inline src/parser/parser.cpp:110]: Should this not be `pos + 1`?

**Comment by lnkuiper** [comment]
I've implemented your feedback, but the compiler is telling me this:
```
/home/runner/work/duckdb/duckdb/src/parser/parser.cpp:47:64: error: comparison is always true due to limited range of data type [-Werror=type-limits]
   47 |  return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '\200' && c <= '\377');
      |                                                              ~~^~~~~~~~~
```
This is the same comparison that is in `rendering.cpp`. Can it just be removed? I've tried with both signed and unsigned char.

File: src/parser/parser.cpp (hunk 1 above)
Comment (lnkuiper) [reply src/parser/parser.cpp]: Yes, thanks!

**Comment by Mytherin** [comment]
> I've implemented your feedback, but the compiler is telling me this:
> 
> ```
> /home/runner/work/duckdb/duckdb/src/parser/parser.cpp:47:64: error: comparison is always true due to limited range of data type [-Werror=type-limits]
>    47 |  return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '\200' && c <= '\377');
>       |                                                              ~~^~~~~~~~~
> ```
> 
> This is the same comparison that is in `rendering.cpp`. Can it just be removed? I've tried with both signed and unsigned char.

\200 (octal) is 128 in decimal, so I don't see how that can be out of range for an unsigned char (which has the domain `0..255`). Are you sure it fails for unsigned char? You might also need to cast the literal, e.g.  `(unsigned char)'\200'`

**Comment by Mytherin** [comment]
But yes good point this might not be handled correctly in rendering.cpp either - we just don't compile that with the same warnings as we do the other code

**Comment by Mytherin** [comment]
Without that check we don't correctly support unicode in tags, e.g.:

```sql
select $ü$xxx$ü$;
```

**Comment by lnkuiper** [comment]
Syntax highlighting seems to work properly in the CLI. I won't touch that code.

I've changed the check to the equivalent hex literals instead:
```c++
return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= 0x80 && c <= 0xFF);
```
And now, the comparison with `0x80` is fine; there is no compiler warning. The compiler is telling me that the comparison with `0xFF` is redundant, though. This makes sense since it's `255`, the absolute max for an `unsigned char`, so I guess I'll remove that.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**